added lowercase version of payload.Message

obo20 opened the pull request:
For some reason, when passing in an invalid CID: the payload returned is: 
```
{
  "code": 400,
  "message": "error decoding Cid: selected encoding not supported"
}
```

Because the request-api expects an uppercase "Message" this causes the error message to look like:

```
{
   message: "[object Object]"
}
```

adding this extra case fixes this issue and correctly returns the error message

obo20 committed:
added lowercase version of payload.Message

For some reason, when passing in an invalid CID: the payload returned is: 
```
{
  "code": 400,
  "message": "error decoding Cid: selected encoding not supported"
}
```

Because the request-api expects an uppercase "Message" this causes the error message to look like:

```
{
   message: "[object Object]"
}
```

adding this extra case fixes this issue and correctly returns the error message

obo20 commented:
@vasa-develop @hsanjuan This may be an issue with cluster itself, but I wanted to throw this PR here as a starting point

hsanjuan commented:
Cluster REST API always returns lowercase json keys but ipfs doesn't. This API is inspired in the js-ipfs one and has carried this issue with it.

hsanjuan commented on src/utils/request-api.js:
```diff
@@ -21,7 +21,7 @@ function parseError (res, cb) {
 
     if (payload) {
       error.code = payload.Code
```
This is probably also `payload.code`

obo20 replied:
good catch

hsanjuan commented on src/utils/request-api.js:
```diff
@@ -21,7 +21,7 @@ function parseError (res, cb) {
       error.code = payload.Code
-      error.message = payload.Message || payload.toString()
+      error.message = payload.message || payload.Message ||payload.toString()
       error.type = payload.Type
```
Cluster will never set an error type, only `code` and `message`.

obo20 replied:
updated to remove type

obo20 committed:
removed type check from error creation

returned the type check from the error creation process as this won't ever be set by cluster

obo20 committed:
added lowercase condition to error formation

added a lowercase check as well to the error formation as cluster returned a lowercase code
